pandas/core/generic.py

```diff
@@ -699,18 +699,100 @@ def pop(self, item):
 
     def squeeze(self, axis=None):
         """
-        Squeeze length 1 dimensions.
+        Squeeze 1 dimensional axis objects into scalars.
+
+        If the given axis consists of one dimensional objects, they are turned
+        into scalars. In case no axis is specified, all axes are subject to
+        squeezing. In any case, objects in axes that can't be squeezed are
+        left unchanged.
```

Comment:
I think it's a good description, but personally I find that being a bit less technical would make it easier to understand. For example "If applied in a 1x1 DataFrame, it returns the contained value. In a DataFrame with one column, it converts it to a Series...". Feel free to disagree, but IMO something like this is a bit faster to understand.

Reply:
The hard part of describing squeeze it's that it's too general, but perhaps I lost touch with practicality because I was thinking of squeezing N-dimensional objects, even though pandas deals mainly with 1D (Series) and 2D (Frames).

I hoped to make it concrete in the examples, but I'm afraid people will just give up reading after a paragraph like this.